### DISCUSSION TITLE
markdown引用图片的合理位置

### DISCUSSION THREAD

**Discussion by raysun0501** [vuepress-theme-hope/vuepress-theme-hope#3318] [Q&A] [answered]
场景描述：
由于博客内容需要大量引用图片(大概1篇博客10多张图片)，有如下困惑：
由于引用图片较多，所以，如果将图片都放置在src/public下面的某文件夹，则博客文章引用时图片时，需要先将图片文件复制“src/publi/某文件夹”中，然后再通过markdown语法([]()) 将图片一个个引入，非常麻烦。
但是通过直接对图片执行复制粘贴操作到markdown文件中，则可以很方便的插入图片，缺点图片位置默认与markdown文件所在目录同级，如果同一目录下有多个markdown文件则会导致不同markdown文件引用的图片位于同一目录下，会导致图片作用域难以管理；
为此，只能为每篇文章建一个文件夹，文件夹中只保存这篇文章对应的markdown以及其引用的图片，但是这样的话，sidebar会将文件夹也解析成为目录，导致通过sidebar查看文章时，每次都需要先点开与文章同名的目录，再点击对应的markdown文章才能查看文章，不够优雅且很繁琐，所以，咨询下是否有较为优雅的、可较好控制作用域的方式放置markdown引用的图片

**Comment by Mister-Hope** [comment, marked as answer]
也许，你可以用permalink来重新处理一下页面的路径？把 `/a/b/` 变为 `/a/b.html`

**Comment by raysun0501** [comment]
“在为每篇文章建一个文件夹，文件夹中只保存这篇文章对应的markdown以及其引用的图片时，@vuepress/plugin-catalog插件会将文件夹也解析成为目录，导致通过navbar查看文章时，每次都需要先点开与文章同名的目录，再点击对应的markdown文章才能查看文章，不够优雅且很繁琐。”
对于这一问题，可通过在index.scss中自定义css样式解决，详见本人仓库的配置
https://github.com/RaySunWHUT/RaySunWHUT.github.io/blob/master/src/.vuepress/styles/index.scss
